src/components/ThemeProvider/__stories__/ThemeProvider.stories.mdx

```diff
@@ -0,0 +1,166 @@
+  ThemeProviderThemingScopeTemplate
+} from "./ThemeProvider.stories";
+import { SystemTheme, ThemeColor } from "../ThemeProviderConstants";
+import { ColorsEligibleForTheming } from "./colors-eligible-for-theming";
+
+export const metaSettings = createStoryMetaSettingsDecorator({
+  component: ThemeProvider,
+  enumPropNamesArray: [], // List enum props here
+  iconPropNamesArray: [], // List props that are typed as icons here
+  actionPropsArray: [] // List the component's actions here
+});
+
+<Meta
+  title="Components/ThemeProvider"
+  component={ThemeProvider}
+  argTypes={metaSettings.argTypes}
+  decorators={metaSettings.decorators}
+/>
+
+# ThemeProvider
+
+<AlphaWarning />
+
+- [Overview](#overview)
+- [Props](#props)
+- [Usage](#usage)
+- [Variants](#variants)
+- [Do’s and don’ts](#dos-and-donts)
+- [Feedback](#feedback)
+
+## Overview
+
+This component helps to customize the colors of library's components by overriding a specific css variables with a values provided within theme object.
+There are 2 levels of theming: **system theme** and **product theme**.
+System theme is a one of a 4 predefined themes: <code>light</code>(<code>.light-app-theme</code>), <code>dark</code>(<code>.dark-app-theme</code>), <code>black</code>(<code>.black-app-theme</code>) and <code>hacker</code>(<code>.hacker_theme-app-theme</code>).
+Product theme is a custom theme that can be provided by a user, there you can specify the values of specific color tokens for each of the system themes.
+
+<Canvas>
+  <Story
+    name="Overview"
+    args={{
+      theme: {
+        name: "overview-theme",
+        colors: {
+          [SystemTheme.LIGHT]: {
+            [ThemeColor.primaryColor]: "green",
+            [ThemeColor.primaryHoverColor]: "darkgreen"
+          },
+          [SystemTheme.DARK]: {
+            [ThemeColor.primaryColor]: "red",
+            [ThemeColor.primaryHoverColor]: "darkred"
+          },
+          black: {
+            "primary-color": "orange",
+            "primary-hover-color": "darkorange"
+          },
+          hacker: {
+            "primary-color": "purple",
+            "primary-hover-color": "darkmagenta"
+          }
+        }
+      }
+    }}
+  >
+    {ThemeProviderTemplateOverview.bind({})}
+  </Story>
+</Canvas>
+
+## Props
+
+<ArgsTable story="Overview" />
+
+## Usage
+
+<UsageGuidelines
+  guidelines={[
+    <>
+      Control themes in your application by setting theme classes (e.g. <code>.light-app-theme</code>) on your{" "}
+      <code>body</code> and render everything else inside it
+    </>,
+    <>
+      In most common case ThemeProvider should be rendered only once on the root level of the application - below the{" "}
+      <code>body</code>
+    </>,
+    <>
+      ThemeProvider is populating theme name <code>className</code> to it's child, so don't put{" "}
+      <code>{"\<Fragment\>"}</code> (<code>{"\<\>"}</code>) inside - it's not accepting <code>className</code> prop
+    </>
```

Comment:
Currently there are problems with displaying this - @talkor will take care to fix it UsageGuidelines component styles

Reply:
Fixed